### ISSUE TITLE
Add `{MISE_CONFIG_DIR}/config.{MISE_ENV}.local.toml` to config lookup

### ISSUE THREAD

**Issue by Drew-Daniels** [jdx/mise#2773]
I have set `MISE_CONFIG_DIR` to `~/projects/dotfiles/mise` and have a `config.local.toml` file with global env vars that I want to make available to all my shells.

However, I see that this file is not getting loaded, and it doesn't look like this file is included in the mise file lookup process, at least as far as I can tell by looking at the [documentation on Profiles](https://mise.jdx.dev/profiles.html#profiles).

I host my `dotfiles` on a remote GitHub repository, and have a `mise/config.toml` file that I have committed with settings I want to share across machines, but I would like to also have a `mise/config.local.toml` file that I do not track that can hold all of my secrets & machine-specific configurations, and while it looks like this is supported if you have `config.local.toml` in a directory like `~/.config/mise`, it doesn't look like you can if you're setting `MISE_CONFIG_DIR` to some other directory.

My `.zshrc` (with `XDG_CONFIG_HOME` set to `~/projects/dotfiles`):
```sh
# mise
export MISE_CONFIG_DIR="$XDG_CONFIG_HOME/mise"
# development, testing, production
export MISE_ENV="development"
eval "$(/opt/homebrew/opt/mise/bin/mise activate zsh)"

export MISE_RUBY_DEFAULT_PACKAGES_FILE="$XDG_CONFIG_HOME/mise/.default-gems"
export MISE_NODE_DEFAULT_PACKAGES_FILE="$XDG_CONFIG_HOME/mise/.default-npm-packages"
export MISE_PYTHON_DEFAULT_PACKAGES_FILE="$XDG_CONFIG_HOME/mise/.default-python-packages"
```

```sh
❯ mise doctor
version: 2024.10.7 macos-arm64 (2024-10-14)
activated: yes
shims_on_path: no

build_info:
  Target: aarch64-apple-darwin
  Features: DEFAULT, NATIVE_TLS
  Built: Mon, 14 Oct 2024 22:13:17 +0000
  Rust Version: rustc 1.81.0 (eeb90cda1 2024-09-04) (Homebrew)
  Profile: release

shell:
  fish
  fish, version 3.7.1

dirs:
  data: ~/.local/share/mise
  config: ~/projects/dotfiles/mise
  cache: ~/Library/Caches/mise
  state: ~/.local/state/mise
  shims: ~/.local/share/mise/shims

config_files:
  ~/projects/dotfiles/mise/config.toml

backends:
  cargo
  core
  go
  npm
  pipx
  spm
  ubi
  vfox

plugins:
  node  https://github.com/asdf-vm/asdf-nodejs#c36e6f0

toolset:
  python@3.12.5
  node@22.5.1
  go@1.21.1
  aws-vault@7.2.0  (missing)
  deno@1.45.2      (missing)
  java@17.0.2
  usage@0.3.0      (missing)
  ruby@3.3.4

env_vars:
  MISE_SHELL=fish
  MISE_NODE_DEFAULT_PACKAGES_FILE=~/projects/dotfiles/mise/default-npm-packages
  MISE_CONFIG_DIR=~/projects/dotfiles/mise
  MISE_ENV=development
  MISE_PYTHON_DEFAULT_PACKAGES_FILE=~/projects/dotfiles/mise/default-python-packages
  MISE_RUBY_DEFAULT_PACKAGES_FILE=~/projects/dotfiles/mise/default-gems

settings:
  activate_aggressive = false
  all_compile = false
  always_keep_download = false
  always_keep_install = false
  asdf_compat = false
  cache_prune_age = "30d"
  ci = false
  color = true
  debug = false
  disable_backends = []
  disable_default_registry = false
  disable_hints = []
  disable_tools = []
  experimental = true
  fetch_remote_versions_cache = "1h"
  fetch_remote_versions_timeout = "10s"
  go_default_packages_file = "~/.default-go-packages"
  go_download_mirror = "https://dl.google.com/go"
  go_repo = "https://github.com/golang/go"
  go_set_gopath = false
  go_set_goroot = true
  go_skip_checksum = false
  http_timeout = "60"
  jobs = 4
  legacy_version_file = true
  legacy_version_file_disable_tools = []
  libgit2 = true
  log_level = "info"
  not_found_auto_install = true
  paranoid = false
  pin = false
  plugin_autoupdate_last_check_duration = "7d"
  python_default_packages_file = "~/.default-python-packages"
  quiet = false
  raw = false
  trace = false
  trusted_config_paths = []
  use_versions_host = true
  verbose = false
  yes = false

  [cargo]
  binstall = true

  [node]

  [pipx]
  uvx = false

  [python]
  default_packages_file = "~/.default-python-packages"
  pyenv_repo = "https://github.com/pyenv/pyenv.git"
  venv_auto_create = false
  venv_stdlib = false

  [ruby]
  default_packages_file = "~/projects/dotfiles/mise/default-gems"
  ruby_build_repo = "https://github.com/rbenv/ruby-build.git"
  ruby_install = false
  ruby_install_repo = "https://github.com/postmodern/ruby-install.git"

  [status]
  missing_tools = "if_other_versions_installed"
  show_env = false
  show_tools = false
```

## TLDR

We have a lookup for `{MISE_CONFIG_DIR}/config.{MISE_ENV}.toml`, I would like `mise` to _also_ lookup `{MISE_CONFIG_DIR}/config.{MISE_ENV}.local.toml`
